opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -118,63 +118,150 @@ message MetricDescriptor {
+    //   8. The 1 second collection cycle ends. A metric is exported for the
+    //      number of requests received over the interval of time t_0+1 to
+    //      t_0+2 with a value of 2.
+    DELTA = 2;
+
+    // CUMULATIVE is a metric whose values are the aggregation of
+    // successively made measurements from a fixed start time until the last
+    // reported measurement. This means that current values of a CUMULATIVE
+    // metric depend on all previous measurements since the start time.
+    // Because of this, the sender is required to retain this state in some
+    // form. If this state is lost or invalidated, the CUMULATIVE metric
+    // values MUST be reset and a new fixed start time following the last
+    // reported measurement time sent MUST be used.
+    //
+    // For example, consider a system measuring the number of requests that
+    // it receives every second and reports the sum of these requests as a
+    // CUMULATIVE metric:
+    //
+    //   1. The system starts receiving at time=t_0.
+    //   2. A request is received, the system measures 1 request.
+    //   3. A request is received, the system measures 1 request.
+    //   4. A request is received, the system measures 1 request.
+    //   5. The 1 second collection cycle ends. A metric is exported for the
+    //      number of requests received over the interval of time t_0 to
+    //      t_0+1 with a value of 3.
+    //   6. A request is received, the system measures 1 request.
+    //   7. A request is received, the system measures 1 request.
+    //   8. The 1 second collection cycle ends. A metric is exported for the
+    //      number of requests received over the interval of time t_0 to
+    //      t_0+2 with a value of 5.
+    //   9. The system experiences a fault and loses state.
+    //   10. The system recovers and resumes receiving at time=t_1.
+    //   11. A request is received, the system measures 1 request.
+    //   12. The 1 second collection cycle ends. A metric is exported for the
+    //      number of requests received over the interval of time t_1 to
+    //      t_0+1 with a value of 1.
+    CUMULATIVE = 3;
+  }
+
+  // temporality is the Temporality of values this metric has.
+  Temporality temporality = 5;
+
+  // Monotonic is a refinement of the values a metric has. It defines the
+  // relationship values of successively reported metrics have
+  // (non-increasing, non-decreasing, or unknown). This is a refinement of
+  // the metric values that can be useful for a receiver in understanding
+  // how to deal with discontinuities in the data (i.e.  calculating
+  // derivates of the data without introducing artifacts from a reset).
+  enum Monotonic {
+    // MONOTONIC_UNSPECIFIED is the default, and means the monotonic nature
+    // of the metric values is unknown.
+    MONOTONIC_UNSPECIFIED = 0;
+
+    // NONDECREASING means all the successive metric values increase or
+    // remain constant.
+    NONDECREASING = 1;
```

Comment:
Does `CUMULATIVE` imply `NONDECREASING`? What does a combination of `CUMULATIVE` and `MONOTONIC_UNSPECIFIED` mean? Should `monotonic` field be ignored for `CUMULATIVE`?

Assuming that in the future `Monotonic` enum may be extended with `NONINCREASING` will it mean that a combination of `CUMULATIVE` and `NONINCREASING` is invalid or that the `CUMULATIVE` metric can count downwards?

Reply:
> Does `CUMULATIVE` imply `NONDECREASING`? What does a combination of `CUMULATIVE` and `MONOTONIC_UNSPECIFIED` mean? Should `monotonic` field be ignored for `CUMULATIVE`?
>
> Assuming that in the future `Monotonic` enum may be extended with `NONINCREASING` will it mean that a combination of `CUMULATIVE` and `NONINCREASING` is invalid or that the `CUMULATIVE` metric can count downwards?

While these concepts are commonly associated they are not coupled. For example, if an instrument measures the egress/ingress ratio of network bytes (with ingress being a positive value and egress being a negative value) and sees a total ingress of 1 byte in one cycle and total egress of 2 byte if the instrument reports this as a cumulative count of bytes it would report `1 Byte` and subsequently `-1 Byte`.

Reply:
I think this should be clarify in a comment that it is different than Stackdriver definition. Don't know if we want to point to a specific product but may be good to call that.